assets/case/reschedule_finish_hearing_deadline.js

```diff
@@ -0,0 +1,21 @@
+/* global $ */
+
+$('#finish_hearing_submit_button').on('click', function (e) {
+  const $form = $('#reschedule_hearing_form')
+  e.preventDefault()
+  // Submit data via AJAX to the form's action path.
+  $.ajax({
+    url: $form.attr('action'),
+    type: $form.attr('method'),
+    data: $form.serialize(),
+    success: function (html) {
+      if ($(html).find('.form-error-message').length) {
+        $('#reschedule_finish_hearing_deadline').modal('hide')
+        $(html).modal('show')
```

Comment:
I suppose you're doing something clever here, but it's not obvious (to me) what actually happens. A comment with a short explanation might come in handy.

Reply:
It seems replacing `$('#reschedule_finish_hearing_deadline')`, the first modal, with the response `html` modal does not do the job.
Hiding the first and showing the response does the job.

Reply:
Actually it does not do the job at all. Causing errors by selecting a date which does not validate causes same issue as before. Looking at this again.

Reply:
Fixed in fe66e0a516dc35b6d759cb463b1b43772aaf1e07.